src/modules/Api.ts

```diff
@@ -183,6 +192,163 @@ export class Api {
     });
   }
 
+  public async placeCanonicalOrder({
+    order: {
+      side,
+      market,
+      amount,
+      price,
+      makerAccountOwner,
+      expiration = new BigNumber(FOUR_WEEKS_IN_SECONDS),
+      limitFee,
+    },
+    fillOrKill,
+    postOnly,
+    clientId,
+    cancelId,
+    cancelAmountOnRevert,
+  }: {
+    order: {
+      side: ApiSide,
+      market: ApiMarketName,
+      amount: BigNumberable,
+      price: BigNumberable,
+      makerAccountOwner: address,
+      expiration: BigNumberable,
+      limitFee?: BigNumberable,
+    },
+    fillOrKill?: boolean,
+    postOnly?: boolean,
+    clientId?: string,
+    cancelId?: string,
+    cancelAmountOnRevert?: boolean,
+  }): Promise<{ order: ApiOrder }> {
+    const order: SignedCanonicalOrder = await this.createCanonicalOrder({
+      side,
+      market,
+      amount,
+      price,
+      makerAccountOwner,
+      expiration,
+      limitFee,
+      postOnly,
+    });
+
+    return this.submitCanonicalOrder({
+      order,
+      fillOrKill,
+      postOnly,
+      cancelId,
+      clientId,
+      cancelAmountOnRevert,
+    });
+  }
+
+  /**
+   * Creates but does not place a signed canonicalOrder
+   */
+  async createCanonicalOrder({
+    side,
+    market,
+    amount,
+    price,
+    makerAccountOwner,
+    expiration,
+    limitFee,
+    postOnly,
+  }: {
+    side: ApiSide,
+    market: ApiMarketName,
+    amount: BigNumberable,
+    price: BigNumberable,
+    makerAccountOwner: address,
+    expiration: BigNumberable,
+    limitFee?: BigNumberable,
+    postOnly?: boolean,
+  }): Promise<SignedCanonicalOrder> {
+    if (!side || !market) {
+      throw new Error(`side: ${side} or market: ${market} is null`);
+    }
+
+    const markets: string[] = market.split('-');
+    const baseMarket: BigNumber = MarketId[markets[0]];
+    const quoteMarket: BigNumber = MarketId[markets[1]];
+    if (!MarketId[markets[0]] || !MarketId[markets[1]]) {
```

Comment:
nit: can check baseMarket and quoteMarket directly

Reply:
actually just check that market is in ApiMarketName since ETH should be considered bad for passing in as a market as well

Reply:
For example, 'ETH-DAI' should throw an error